### DISCUSSION TITLE
Using one Tasmota device as a remote switch to switch a relay on a second device

### DISCUSSION THREAD

**Discussion by Stefhorn** [arendst/Tasmota#21701] [Q&A]
Hi all
I don't have much experience with Tasmota devices and have only used them to switch power sockets on and off.
Now I would like to use 2 Tasmota units (without a hub) as follows:
2 push buttons are connected to device A and a relay is connected to device B. Both devices are physically separated but on the same Wi-Fi network. 
When device B is powered up, the relay turns on. To switch the relay off the Off button needs to be pressed for >10 seconds. When the Off button is released, the relay must remain off. To switch the relay on again, the ON button must be pressed. As the switching device is battery powered, it needs to go into deep sleep mode after switching. 
Is this technically possible and how is it done?

**Comment by ferbulous** [comment]
I think you can have a look at tasmesh,
The relay needs to be on esp32 as broker

**Reply by sfromis** [reply]
For communication between units, it may be simpler to use the Device Groups feature, or the `WebSend` command.

**Reply by ferbulous** [reply]
Does device group still work with deep sleep?

**Reply by sfromis** [reply]
If the devices both are on the network (Wifi), device groups should work (UDP communication). That is unrelated to deep sleep at other times.

**Reply by joba-1** [reply]
I guess it depends on how device groups handle a device waking up: will it receive the current status if it slept during the last status change?

Generally there are quite some corner cases around devices rebooting or going to sleep and waking up and possible status changes of the other device during these times. But not that many. Thnink them through and decide what behaviour you would like to see and test what you get. Everything is tweakable...

**Reply by sfromis** [reply]
UDP (hence Device Groups) is fire-and forget, nothing like actions being queued for inactive devices. However, emitting actions **from** the battery device should not depend on such.

**Reply by ferbulous** [reply]
I haven’t tried deep sleep yet
Let’s say I use a button (gpio16), to wake the device -> boots up -> connect to wifi (I assume by this point it’s joined the device group)

Then i could set a rule, like on wifi do relay toggle (which gets sent to the group), then resume to deep sleep. Would that work?

**Reply by sfromis** [reply]
Well, try it :wink: 

At first, without bothering with deep sleep or rules.

**Comment by Stefhorn** [comment]
Thanks guys for all your input. On my return from holiday I will have a closer look at your suggestions and report back.

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________
